lib/domain/goal.js

```diff
@@ -0,0 +1,19 @@
+export default class Goal {
+  constructor({ targetReviewDate, text, useAsPhp }) {
+    this.agreedDate = this.todaysDate();
+    this.completedDate = null;
+    this.targetReviewDate = this.reviewDate(targetReviewDate);
+    this.text = text;
+    this.useAsPhp = useAsPhp;
+  }
+
+  reviewDate = date => {
+    return new Date(date.year, date.month - 1, date.day).toISOString();
+  };
+
+  todaysDate = () => {
+    const today = new Date();
+    today.setUTCHours(0, 0, 0, 0);
+    return today.toISOString();
+  };
```

Comment:
At the moment these are exposed as methods on this class, so if I have a goal I could call `myExcitingGoal.todaysDate()` which seems a bit like we are polluting that API with internal logic?

Since they don't depend on instance members you can hide these by just moving them outside of the class, and you'll still be able to access them.

```
const reviewDate = date => { ...  };
const todaysDate = () => { ... };

export default class Goal {
  constructor(...) {
    this.targetReviewDate = reviewDate(targetReviewDate);
  }
...
};
```